sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3907,8 +4016,10 @@ pi_result piQueueFinish(pi_queue Queue) {
     }
   }
   // Reset signalled command lists and return them back to the cache of
-  // available command lists.
-  resetCommandLists(Queue);
+  // available command lists. Events in the immediate command lists are cleaned
+  // up in synchronize().
```

Comment:
> Events in the immediate command lists are cleaned up in synchronize().

Could we move that code here instead?

Reply:
In current implementation it will be worse if I move that code here. synchronize() is done under queue lock and I just clean all event from the queue after synchronization there.  This code is not under queue lock and I won't be able to simply clean all events.

Reply:
OK